low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,500 @@
+            msg = "Facing problem while creating sspl support bundle : %s" % err
+            self.__clear_tmp_files()
+            raise SupportBundleError(1, msg)
+        try:
+            # Process request
+            BundleRequestHandler._update_request_status(
+                request_id, BundleRequestHandler.SUCCESS)
+        except Exception as err:
+            status = BundleRequestHandler.FAILED + f" - {err}"
+            BundleRequestHandler._update_request_status(
+                request_id, status)
+        return request_id
+
+    def get_ipmi_sel_data(self):
+        ipmitool = IpmiFactory().get_implementor('ipmitool')
+        res, err, retcode = ipmitool._run_ipmitool_subcommand("sel list")
+        if retcode != 0:
+            if err.find(ipmitool.VM_ERROR) != -1:
+                err = ("Detected VM environment, can't reach BMC over inband"
+                       "KCS channel, avoiding BMC data collection using IPMI.")
+            logger.warn(
+                f"Unable to fetch SEL list data with an error : {err}")
+        else:
+            with open(self.ipmi_sel_data, 'w+') as sel_list:
+                sel_list.write(res)
+
+    def get_config_data(self):
+        os.makedirs(self.sspl_conf_dir, exist_ok=True)
+        for conf_file in glob.glob('/etc/sspl*.*'):
+            shutil.copy(conf_file, self.sspl_conf_dir)
+
+    def _run_command(self, command):
+        """Run the command and get the response and error returned"""
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+        return_code = process.wait()
+        response, error = process.communicate()
+        return response.rstrip('\n'), error.rstrip('\n'), return_code
+
+    def get_drives_smart_data(self):
+        lsscsi_cmd = " ".join(["lsscsi", "|", "grep", "disk"])
+        lsscsi_cmd = shlex.split(lsscsi_cmd)
+        # lsscsi_response, _, _ = SimpleProcess(lsscsi_cmd).run()
+        lsscsi_response, _, _ = self._run_command(lsscsi_cmd)
+        os.makedirs(self.boot_drvs_dta, exist_ok=True)
+        for res in lsscsi_response.split("\n"):
+            drive_path = res.strip().split(' ')[-1]
+            smartctl_cmd = f"sudo smartctl -a {drive_path} --json"
+            # response, _, _ = SimpleProcess(smartctl_cmd).run()
+            response, _, _ = self._run_command(smartctl_cmd)
+            response = json.loads(response)
+            try:
+                if 'device' in response and response['device']['protocol'] == 'ATA':
+                    file_name = drive_path.replace('/', '_')[1:]
+                    with open(f"{self.boot_drvs_dta}/{file_name}.json", "w+") as fp:
+                        json.dump(response, fp,  indent=4)
+
+            except Exception as e:
+                logger.error(
+                    "Error in writing {0} file: {1}".format(response, e))
+
+    def get_enclosure_logs(self):
```

Comment:
why we are duplicating code? These functions can be easily put under some utils class, like here StorageEncl(), etc

Reply:
Moved it to `StorageEncl` class.